packages/medusa/src/models/category.ts

```diff
@@ -0,0 +1,112 @@
+import { generateEntityId } from "../utils/generate-entity-id"
+import { SoftDeletableEntity } from "../interfaces/models/soft-deletable-entity"
+import {
+  BeforeInsert,
+  Index,
+  Entity,
+  Tree,
+  Column,
+  PrimaryGeneratedColumn,
+  TreeChildren,
+  TreeParent,
+  TreeLevelColumn,
+  JoinColumn,
+} from "typeorm"
+
+@Entity()
+@Tree("materialized-path")
+export class Category extends SoftDeletableEntity {
+  @Column()
+  name: string
+
+  @Index({ unique: true, where: "deleted_at IS NULL" })
+  @Column({ nullable: false })
+  handle: string
+
+  @Column()
+  is_active: Boolean
+
+  @Column()
+  is_internal: Boolean
+
+  // The materialized path column is added dynamically by typeorm. Commenting this here for it
+  // to not be a mystery
+  // https://github.com/typeorm/typeorm/blob/62518ae1226f22b2f230afa615532c92f1544f01/src/metadata-builder/EntityMetadataBuilder.ts#L615
+  // @Column({ nullable: true, default: '' })
+  // mpath: String
```

Comment:
thought: Do you think we should maintain the materialized path and parent id ourselves to not rely on Typeorm? I realize we won't change ORM in the near future, but say we want to do that at some point, it might be an easier migration for us if we've handled the logic ourselves. 

This highly depends on how much logic we would need to "replicate", but as far as I can see it's not a lot.

Reply:
I think we should leverage the native API for now and maybe re discuss that point in a later future as it is localised.

Reply:
Yeah, I agree. That's the best approach for now. Discard my comment :)

Reply:
Sounds good! 💪